modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/portlet_url/create_url.es.js

```diff
@@ -0,0 +1,124 @@
+import {isObject, isDef, isNull} from 'metal';
+
+export default function createURL(basePortletURL, parameters) {
+	let internalParameters = {};
+
+	let reservedParameters = {
+		doAsGroupId: null,
+		doAsUserId: null,
+		doAsUserLanguageId: null,
+		p_auth: null,
+		p_auth_secret: null,
+		p_f_id: null,
+		p_j_a_id: null,
+		p_l_id: null,
+		p_l_reset: null,
+		p_p_auth: null,
+		p_p_cacheability: null,
+		p_p_i_id: null,
+		p_p_id: null,
+		p_p_isolated: null,
+		p_p_lifecycle: null,
+		p_p_mode: null,
+		p_p_resource_id: null,
+		p_p_state: null,
+		p_p_state_rcv: null,
+		p_p_static: null,
+		p_p_url_type: null,
+		p_p_width: null,
+		p_t_lifecycle: null,
+		p_v_l_s_g_id: null,
+		refererGroupId: null,
+		refererPlid: null,
+		saveLastPath: null,
+		scroll: null
+	};
+
+	let options = {
```

Comment:
Do we need these options? 
- If they are settable from outside, they need to be one of the arguments. `createURL(basePortletURL, parameters, options)`. Then, we need a `const DEFAULT_OPTIONS`, etc.
- If they are not settable from outside, any code related to them can be removed..